mava/systems/ppo/sebulba/ff_ippo.py

```diff
@@ -11,3 +11,811 @@
+    assert (
+        config.arch.num_envs % len(config.arch.learner_device_ids) == 0
+    ), "The number of environments must to be divisible by the number of learners "
+
+    assert (
+        int(config.arch.num_envs / len(config.arch.learner_device_ids))
+        * config.arch.n_threads_per_executor
+        % config.system.num_minibatches
+        == 0
+    ), "int(local_num_envs / len(learner_device_ids)) must be divisible by num_minibatches"
+
+    # Setup learner.
+    learn, apply_fns, learner_state = learner_setup(
+        (key, actor_net_key, critic_net_key), config, learner_devices
+    )
+
+    # Generate Numpy RNG for reproducibility
+    np_rng = np.random.default_rng(config.system.seed)
+
+    # Setup evaluator.
+    evaluator, absolute_metric_evaluator = make_eval_fns(
+        environments.make_gym_env, apply_fns[0], config, np_rng
+    )
+
+    # Calculate total timesteps.
+    config = sebulba_check_total_timesteps(config)
+    assert (
+        config.system.num_updates > config.arch.num_evaluation
+    ), "Number of updates per evaluation must be less than total number of updates."
+    # Calculate number of updates per evaluation.
+    config.system.num_updates_per_eval = config.system.num_updates // config.arch.num_evaluation
+    config.arch.num_evaluation, remaining_updates = divmod(
+        config.system.num_updates, config.system.num_updates_per_eval
+    )
+    config.arch.num_evaluation += (
+        remaining_updates != 0
+    )  # Add an evaluation step if the num_updates is not a multiple of num_evaluation
+    steps_per_rollout = (
+        len(config.arch.executor_device_ids)
+        * config.arch.n_threads_per_executor
+        * config.system.rollout_length
+        * config.arch.num_envs
+        * config.system.num_updates_per_eval
+    )
+
+    # Logger setup
+    logger = MavaLogger(config)
+    cfg: Dict = OmegaConf.to_container(config, resolve=True)
+    cfg["arch"]["devices"] = jax.devices()
+    pprint(cfg)
+
+    # Set up checkpointer
+    save_checkpoint = config.logger.checkpointing.save_model
+    if save_checkpoint:
+        checkpointer = Checkpointer(
+            metadata=config,  # Save all config as metadata in the checkpoint
+            model_name=config.logger.system_name,
+            **config.logger.checkpointing.save_args,  # Checkpoint args
+        )
+
+    # Executor setup and launch.
+    unreplicated_params = flax.jax_utils.unreplicate(learner_state.params)
+    params_queues: List = []
+    rollout_queues: List = []
+
+    for _d_idx, d_id in enumerate(  # Loop through each executor device
+        config.arch.executor_device_ids
+    ):
+        # Replicate params per executor device
+        device_params = jax.device_put(unreplicated_params, devices[d_id])
+        # Loop through each executor thread
+        for _thread_id in range(config.arch.n_threads_per_executor):
+            seeds = np_rng.integers(np.iinfo(np.int64).max, size=config.arch.num_envs).tolist()
```

Comment:
Could potentially be too big for jax, as usually jax is only up to 32 bit precision
```suggestion
            seeds = np_rng.integers(np.iinfo(np.int32).max, size=config.arch.num_envs).tolist()
```